File: src/dotnet/Ansight/Pairing/Models/PairingTrust.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Trust-policy metadata associated with a host-issued pairing config.
/// </summary>
public sealed class PairingTrust
{
    /// <summary>
    /// Host-defined trust mode identifier.
    /// </summary>
    public required string Mode { get; set; }

    /// <summary>
    /// Indicates whether the first successful pair must also present the one-time token.
    /// </summary>
    public required bool RequireTokenOnFirstPair { get; set; }

    /// <summary>
    /// Indicates whether LAN discovery is allowed for this config.
    /// </summary>
    public required bool AllowLanDiscovery { get; set; }
}

File: src/ios/Sources/AnsightKit/PairingTrust.swift
import CryptoKit
import Foundation

public struct PairingTrust: Sendable, Codable, Equatable {
    public var mode: String
    public var requireTokenOnFirstPair: Bool
    public var allowLanDiscovery: Bool

    public init(mode: String, requireTokenOnFirstPair: Bool, allowLanDiscovery: Bool) {
        self.mode = mode
        self.requireTokenOnFirstPair = requireTokenOnFirstPair
        self.allowLanDiscovery = allowLanDiscovery
    }
}
